src/modelio.cs

```diff
@@ -90,6 +90,27 @@ interface MDLAsset : NSCopying
 		[Export ("canExportFileExtension:")]
 		bool CanExportFileExtension (string extension);
 
+		[iOS (10,3), TV (10,2), Mac (10,12,4)]
+		[Export ("components", ArgumentSemantic.Copy)]
+		IMDLComponent[] Components { get; }
+
+		[Internal]
```

Comment:
maybe` [Advanced]` but not `[Internal]`. The main reason is that `Type` is easier (and type safe) to use, but it would be costly in a loop, e.g.

```
foreach (var component in list)
   x.SetComponent (type);
```

vs

```
var p = new Protocol (type);
foreach (var component in list)
   x.SetComponent (p);
```

It also makes it possible to  override the method in a subclass.